### PR TITLE
[Enhancement] Java template and SCA configuration

### PR DESCRIPTION
**Pull request by kloessst**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I documented the Java code using JavaDoc style.

### Motivation and Context

Improves the default static code analysis configuration and provides examples for instructors how to configure the tools. Fixes static code analysis findings in the templates.

### Description

- Add config xml files for all tools
- On exercise creation, copy the configs to the test repository if static code analysis is enabled
- Fix issues in templates

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Create programming exercises with static code analysis (one with eclipse and one with maven template)
4. Check the test repository
--> SCA XML configs should be copied to the staticCodeAnalysisConfig directory
5. Check the template and solution repository
--> Template: 0 Issues, Solution: 0 Issues
6. Create a programming exercise without SCA
7. Check the test repository
--> the directory staticCodeAnalysisConfig should not exist

### Test Coverage

Nothing should have changed

### CODE REVIEW

File: src/main/resources/templates/java/eclipse/exercise/src/${packageNameFolder}/Client.java [hunk 1]
```diff
@@ -15,17 +14,27 @@
...
+    }
+
     /**
      * Main method.
      * Add code to demonstrate your implementation here.
+     *
+     * @param args command line arguments
      */
-    public static void main(String[] args) throws ParseException {
+    public static void main(final String[] args) throws ParseException {
```
Comment (MaisiKoleni) [inline src/main/resources/templates/java/eclipse/exercise/src/${packageNameFolder}/Client.java:32]:
```suggestion
    public static void main(String[] args) throws ParseException {
```

File: src/main/resources/templates/java/eclipse/exercise/src/${packageNameFolder}/Client.java [hunk 2]
```diff
@@ -2,9 +2,8 @@
 
 import java.text.*;
 import java.util.*;
-import java.security.SecureRandom;
```
Comment (MaisiKoleni) [inline src/main/resources/templates/java/eclipse/exercise/src/${packageNameFolder}/Client.java:5]:
```suggestion
import java.util.concurrent.ThreadLocalRandom;

```

File: src/main/resources/templates/java/eclipse/exercise/src/${packageNameFolder}/Client.java [hunk 3]
```diff
@@ -58,32 +71,44 @@ public static void main(String[] args) throws ParseException {
...
-     * Creates a random Long within given Range
+     * Creates a random long within the given range.
+     *
+     * @param low the lower bound
+     * @param high the upper bound
+     * @return random long within the given range
      */
     private static long randomLongWithin(long low, long high) {
-        return low + random.nextLong() * (high - low);
+        return low + (long) (Math.random() * (high - low));
```
Comment (MaisiKoleni) [inline src/main/resources/templates/java/eclipse/exercise/src/${packageNameFolder}/Client.java:94]:
```suggestion
        return ThreadLocalRandom.current().nextLong(low, high + 1);
```

File: src/main/resources/templates/java/eclipse/exercise/src/${packageNameFolder}/Client.java [hunk 4]
```diff
@@ -58,32 +71,44 @@ public static void main(String[] args) throws ParseException {
...
-     * Creates a random Integer within given Range
+     * Creates a random int within the given range.
+     *
+     * @param low the lower bound
+     * @param high the upper bound
+     * @return random int within the given range
      */
     private static int randomIntegerWithin(int low, int high) {
-        return low + random.nextInt() * (high - low);
+        return low + (int) (Math.random() * (high - low));
```
Comment (MaisiKoleni) [inline src/main/resources/templates/java/eclipse/exercise/src/${packageNameFolder}/Client.java:105]:
```suggestion
        return ThreadLocalRandom.current().nextInt(low, high + 1);
```

File: src/main/resources/templates/java/eclipse/solution/src/${packageNameFolder}/Client.java [hunk 5]
```diff
@@ -3,11 +3,22 @@
 import java.text.*;
 import java.util.*;
 
-public class Client {
+public final class Client {
```
Comment (MaisiKoleni) [inline src/main/resources/templates/java/eclipse/solution/src/${packageNameFolder}/Client.java:6]: Same for the solution

**Review by fde312 — DISMISSED** [review]
Eclipse Programming Exercises:
SCA XML configuration:
image

Activated all SCA categories and the solution template shows only that one issue is left:
image

For Maven Programming Exercises:
SCA XML configuration:
image

Activated all SCA categories and the solution template shows only that one issue is left:
image

No SCA exercise:
image

Worked also when adding back some commented out checks. Really cool feature! Good job 👍 
image

**Review by fde312 — DISMISSED** [review]
The null pointer fix in `6c5bf67`, solved the last issue in the solution repo:
image

**Review by FrankeLukas — APPROVED** [review]
Lgtm.

**Review by MaisiKoleni — APPROVED** [review]
Works well and code looks good to me (no issues found). SCA still works. Tested on TS1.

**Review by balazs-czopf — APPROVED** [review]
Tested it on TS1, worked as expected. Nice job!

**Review by philippbauch — DISMISSED** [review]
Tested with the help of @MaisiKoleni and @fde312, seems to work fine 👍

**Review by krusche — APPROVED** [review]

**Review by MaisiKoleni — APPROVED** [review]

**Review by krusche — APPROVED** [review]

**Pull request merged**